com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -129,6 +129,10 @@ private void unbindService(CloudControllerClient client, String serviceName) {
         logBindings(bindings);
         for (CloudServiceBinding binding : bindings) {
             CloudApplication application = StepsUtil.getBoundApplication(client.getApplications(), binding.getAppGuid());
+            if (application == null) {
+                throw new IllegalStateException(

```

Comment:
Why not put the exception inside the method by calling `.orElseThrow()` on the Optional there?

Reply:
Also thought of this... the reason I did it this way is because the methods in StepsUtils are not throwing some exceptions(as far as I saw) and decided to do it here?